ext/opcache/Optimizer/zend_inference.c

```diff
@@ -2394,6 +2394,7 @@ static int zend_update_type_info(const zend_op_array *op_array,
 		case ZEND_ISSET_ISEMPTY_STATIC_PROP:
 		case ZEND_ASSERT_CHECK:
 		case ZEND_IN_ARRAY:
+		case ZEND_ARRAY_KEY_EXISTS:
 			UPDATE_SSA_TYPE(MAY_BE_FALSE|MAY_BE_TRUE, ssa_ops[i].result_def);
```

Comment:
array_key_exists() can also return null for the zpp failure case.

Additionally, ARRAY_KEY_EXISTS needs to be handled in https://github.com/php/php-src/blob/master/ext/opcache/Optimizer/sccp.c#L1328. Looking at the ISSET_ISEMPTY_DIM_OBJ case should provide inspiration here.